src/main/java/yeelp/distinctdamagedescriptions/util/lib/damagecalculation/DDDCombatTracker.java

```diff
@@ -132,7 +126,9 @@ public void handleHurtStage(LivingHurtEvent evt) {
 				this.results.hasEffectiveShield(m);
 				this.ctx.getShield().ifPresent((stack) -> stack.damageItem((int) (evt.getAmount() * (this.getRecentResults().getShieldRatio().getAsDouble())), this.getFighter()));
 			});
-			this.type = m.keySet().stream().collect(Collectors.toList()).get(rand.nextInt(m.size()));
+			if(!m.isEmpty()) {
+				this.type = m.keySet().stream().collect(Collectors.toList()).get(rand.nextInt(m.size()));
```

Comment:
Reverting this change is fine, but as it was, I realize it is just as inefficient as creating a new ArrayList. I can implement the `Stream.skip` or Iterator approach at a later date (honestly, it probably isn't enough for a performance impact).

Reply:
Yes, it's more possible to find impacts from rendering or random massive tick time entities or TEs. I think my PR is done here.